lib/vtls/darwinssl.c

```diff
@@ -1996,6 +1996,85 @@ static int verify_cert(const char *cafile, struct Curl_easy *data,
   }
 }
 
+static CURLcode pkp_pin_peer_pubkey(struct SessionHandle *data,
+                                    SSLContextRef ctx,
+                                    const char *pinnedpubkey)
+{  /* Scratch */
+  size_t pubkeylen, realpubkeylen, rsaHeaderLength = 24;
+  unsigned char *pubkey = NULL, *realpubkey = NULL, *rsaHeader = NULL;
+
+  /* Result is returned to caller */
+  CURLcode result = CURLE_SSL_PINNEDPUBKEYNOTMATCH;
+
+  /* if a path wasn't specified, don't pin */
+  if(!pinnedpubkey)
+    return CURLE_OK;
+
+
+  if(!ctx)
+    return result;
+
+  do {
+    SecTrustRef trust;
+    OSStatus ret = SSLCopyPeerTrust(ctx, &trust);
+    if(ret != noErr)
+        break;
+    SecKeyRef keyRef = SecTrustCopyPublicKey(trust);
```

Comment:
Again, per the copy rule, you need to `CFRelease` `keyRef` when you're done with it: right now you don't.